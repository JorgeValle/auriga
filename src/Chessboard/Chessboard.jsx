import React from 'react';

// Components should represent one piece of data model
class Chessboard extends React.Component {

  // It only has a render method, because this is a static version of app
  // To make UI interactive, we need to be able to trigger changes to underlying data model
  render() {
    return (
      <main className="chessboard">
        <Row></Row>
        <Row></Row>
        <Row></Row>
        <Row></Row>
        <Row></Row>
        <Row></Row>
        <Row></Row>
        <Row></Row>
        <Row></Row>
        <Row></Row>
      </main>
    );
  }
}

export default Chessboard;