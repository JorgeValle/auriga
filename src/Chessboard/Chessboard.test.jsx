import React from 'react';
import Adapter from 'enzyme-adapter-react-16';
import { shallow, configure } from 'enzyme';
import Chessboard from './Chessboard.jsx';

configure({
  adapter: new Adapter()
});

describe('<Chessboard />', () => {

  // We shallow render to make sure changes in children don't cause cascading fails
  it('shallow renders', () => {
    const wrapper = shallow(<Chessboard />);
  });

  // Snapshot matching
  it('matches snapshot', () => {
    const wrapper = shallow(<Chessboard />);
    expect(wrapper).toMatchSnapshot();
  });

});

