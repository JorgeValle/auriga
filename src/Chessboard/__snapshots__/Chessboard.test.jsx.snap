// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Chessboard /> matches snapshot 1`] = `ShallowWrapper {}`;
